src/crimson/net/io_handler.h

```diff
@@ -255,7 +255,7 @@ class IOHandler final : public ConnectionHandler {
   class shard_states_t {
   public:
     shard_states_t(seastar::shard_id _sid, io_state_t state)
-      : sid{_sid}, io_state{state} {}
+      : sid{_sid}, io_state{state}, gate{_sid} {}
```

Comment:
Please keep the original comment suggested by @cyx1231st here or in the dedicated constructor as this is non-intuitive.

Reply:
This commit should also include the newly added ctor

Reply:
Typo in the commit message, what do you think on:
```
IOHandler::to_new_sid provides a mechanism to switch to a new core from the current one used.
This behavior is tricky for making sure that we keep the IOHandler::shard_states_t::gate instance to be used on the same core that it was created.
In order to resolve this, we should be able to construct Gated out of specified core_id (and not from this_shard_id).

See: https://github.com/ceph/ceph/pull/58986#discussion_r1733782952
```

The Fixes does not belong here as it resolves other bug.